### DISCUSSION TITLE
Alpine's `x-mask` allows 'invisible' trailing character when used with `wire:model`

### DISCUSSION THREAD

**Discussion by jeffchown** [livewire/livewire#7100] [7. Bugs (archived)]
### Livewire version

Livewire v3.0.10, Alpine v3.13.2, x-mask v3.13.2

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Hello,

I've been implementing x-mask in one of my applications and have recently noticed that my inputs using `x-mask` allow the user to type a trailing character that is not shown in the input but is stored in the property linked via `wire:model`.

For example, using markup:

```
<input
    wire:model="form.phone"
    x-mask="(999)999-9999"
    placeholder="(999)999-9999"
/>
<div>{{ $form->phone }}</div>
      
```

if the user types `1a2b34567890` in the input, then x-mask properly constrains & formats the input to `(123)456-7890` and `$form->phone` echos the same. If the user then continues to type and enters any character (number, letter, etc.), the character, e.g. `x` **is not** echoed in the input, but **is** stored in the `form.phone` property and echoed in the `div` as `(123)456-7890x`(and therefore saved to the db, as e.g. `(123)456-7890x`, upon update).

I have tried many different masks in x-mask and the result is the same for all, they all allow a trailing character.

I did confirm when using 'pure' Alpine w/ `x-mask` (no Livewire), via the following snippet, that all works as expected and **no** trailing character is allowed:

```
 <div x-data="{ message: '' }">
      <input type="text" x-model="message" x-mask="(999)999-9999">

      <span x-text="message">
 </div>
````

When testing, any trailing characters were not added/shown in the message property when using only Alpine.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by DaviMenezes** [comment]
The same here, i'm using to capture modification value to calculate in another moment, and error occur when try $my_value ("2s") + $another_value,. "string + int" = A non-numeric value encountered

**Comment by jeffchown** [comment]
@calebporzio @joshhanley I'm happy to help track this one down if either of you can point me in the right direction.

**Comment by jeffchown** [comment]
Has anyone found a fix for this?  
Until I've found a fix, I can't use `x-mask` without adding some extra/redundant validation before saving to the database.  Thanks.

**Comment by jeffchown** [comment]
UPDATE: I did some digging and, after a few tests/experiments (and a hunch), I determined that adding `.debounce` to any `wire:model` to which I'm applying an `x-mask` 'fixes' the problem.

Therefore, instead of:

`<input type="text" wire:model="phone" x-mask="(999)999-9999" placeholder="(999)999-9999">`

using:

`<input type="text" wire:model.debounce="phone" x-mask="(999)999-9999" placeholder="(999)999-9999">`

properly masks the user input and does *not* allow a trailing (and hidden) character to be added to the component property (in this case `phone`).

This is obviously a workaround and I plan to do some further digging when I have more time, but if anyone has any ideas re: why this works and how we might fix it so the explicit `.debounce` modifier isn't necessary, please let me know.

**Reply by jeffchown** [reply]
Hi @calebporzio @joshhanley - any insights into this item?  

While the `.debounce` modifier is a workaround, it is definitely not ideal.  Add to that the fact that some people using `x-mask` with `wire:model` might not know they are sometimes storing a trailing character in their database tables, and I'd like to find a more complete solution.  

Happy to help with a PR if you can share any ideas/insights you might have that could point me in the right direction.

**Reply by chrismcintosh** [reply]
Debounce also seems to mitigate the issue for me as well.

I'm using x-mask on a credit card field. After upgrade from Livewire 2 to Livewire 3 my credit card field was always failing validation. Debounce seems to be the only answer at the moment.

Thanks @jeffchown

**Reply by jeffchown** [reply]
You're welcome, @chrismcintosh

**Comment by luanfreitasdev** [comment]
Hello! I submitted the failed test for this issue. Thanks

https://github.com/livewire/livewire/pull/7423
